### PR TITLE
Compression for the `INT128` type

### PR DESCRIPTION
**Pull request by nickgerrets**
Solves #6531.

This PR implements a custom bitpacking implementation specifically for 128-bit integers. It is tested using the constant, constant delta, frame of reference and delta frame of reference compression types.

### CODE REVIEW

**Review by Tishj — CHANGES_REQUESTED** [review]
Thanks for the PR!

It looks great, I don't have much to comment on other than a couple minor things.
I was wondering if we might want to add a benchmark for the hugeint compression as well (or add it to existing bitpacking benchmarks)

File: src/storage/compression/bitpacking.cpp [hunk 1]
```diff
@@ -262,7 +269,7 @@ struct BitpackingState {
 		}
 
 		if (can_do_for) {
-			auto width = BitpackingPrimitives::MinimumBitWidth<T_U>(min_max_diff);
+			auto width = BitpackingPrimitives::MinimumBitWidth<T, false>(min_max_diff);
```
[the remark this answers is not in the document]
Comment (nickgerrets) [reply src/storage/compression/bitpacking.cpp:272]: MinimumBitWidth is only called once without `is_signed=false`.

[the remark this answers is not in the document]
Comment (Tishj) [reply src/storage/compression/bitpacking.cpp:272]: Ah never mind then, thanks for checking

**Review by Mytherin — APPROVED** [review]
Thanks for the PR! Looks great! Perhaps @samansmink can also still have a look?

**Review by samansmink — APPROVED** [review]
Impressive work Nick! I have nothing to add code-wise.

I did do a bit of benchmarking by modifying the micro benchmarks in `benchmark/micro/compression/bitpacking` to compare the performance between 
Int32/int64 compression/decompression and hugeints. The rough summary is:
- compression/writing of hugeints is 2-4x slower than int32/64
- descompression/scanning is 4-6x slower than int32/64 and roughly 2-3x slower than uncompressed hugeints

After some quick profiling i think the cause is mainly in the hugeint operator performance. For example for `benchmark/micro/compression/bitpacking/bitpacking_read_dfor.benchmark`, by far most of the time spent in scanning is spent in the hugeint addition operator. An optimization idea there could be to ensure the operators as used during compression don't do overflow checks.  Another optimization might be to down cast to int64 where possible so that regular bitpacking can be used.

However, I think these performance figures are definitely good enough for this PR and any performance optimizations can come in a follow up PR!

**Comment by Mytherin** [comment]
Let's get this in then :) thanks again!

**Pull request merged**